tests/checkers/unittest_imports.py

```diff
@@ -114,3 +114,26 @@ def test_wildcard_import_non_init(self) -> None:
         )
         with self.assertAddsMessages(msg):
             self.checker.visit_importfrom(import_from)
+
+    @staticmethod
+    def test_preferred_module(capsys: CaptureFixture[str]) -> None:
+        """
+        Tests preferred-module configuration option
+        """
+
+        # test preferred-modules case with base module import
+        Run(
+            [
+                f"{os.path.join(REGR_DATA, 'preferred_module')}",
+                "-d all",
+                "-e preferred-module",
+                # prefer sys instead of os (for triggering test)
+                "--preferred-modules=os:sys",
+            ],
+            exit=False,
+        )
+        output, errors = capsys.readouterr()
+
+        # assert that we saw preferred-modules triggered
+        assert len(output.split("\n")) == 7
```

Comment:
Could you assert for the message here? If we ever change the output format this will unnecessarily fail.

Reply:
Thank you for the feedback here! I've made a change to this effect to address this in [4781b0b](https://github.com/PyCQA/pylint/pull/8093/commits/4781b0b96c26865c5a18d6281a01be37c78c95e4).